### PR TITLE
[ISSUE-367][FEATURE] Master support worker re-register with same port and worker miss heartbeat only delete expired data

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### How was this patch tested?

/cc @related-reviewer

/assign @main-reviewer

### CODE REVIEW

**Comment by AngersZhuuuu** [comment]
ping @waitinfuture @FMX

File: server-worker/src/main/scala/com/aliyun/emr/rss/service/deploy/worker/Worker.scala [hunk 1]
```diff
@@ -194,16 +194,16 @@ private[deploy] class Worker(
       response.expiredShuffleKeys.asScala.foreach(shuffleKey => workerInfo.releaseSlots(shuffleKey))
       cleanTaskQueue.put(response.expiredShuffleKeys)
     } else {
-      logError("Worker not registered in master, clean all shuffle data and register again.")
+      logError("Worker not registered in master, clean expired shuffle data and register again.")
       // Clean all shuffle related metadata and data
-      cleanup(shuffleKeys)
+      cleanup(response.expiredShuffleKeys)
```
Comment (waitinfuture) [inline server-worker/src/main/scala/com/aliyun/emr/rss/service/deploy/worker/Worker.scala:199]: shuffleKeys defined above can be deleted

Comment (AngersZhuuuu) [reply server-worker/src/main/scala/com/aliyun/emr/rss/service/deploy/worker/Worker.scala:199]: > shuffleKeys defined above can be deleted

Used in `Heatbeat` message

File: server-master/src/main/scala/com/aliyun/emr/rss/service/deploy/master/Master.scala [hunk 2]
```diff
@@ -366,16 +365,18 @@ private[deploy] class Master(
...
     if (workersSnapShot.contains(workerToRegister)) {
       logWarning(s"Receive RegisterWorker while worker" +
-        s" ${workerToRegister.toString()} already exists,trigger WorkerLost.")
-      if (!statusSystem.workerLostEvents.contains(workerToRegister)) {
-        self.send(WorkerLost(host, rpcPort, pushPort, fetchPort, replicatePort,
-          RssHARetryClient.genRequestId()))
-      }
-      context.reply(RegisterWorkerResponse(false, "Worker already registered!"))
+        s" ${workerToRegister.toString()} already exists, re-register.")
+      statusSystem.workers.remove(workerToRegister)
```
[the remark this answers is not in the document]
Comment (AngersZhuuuu) [reply server-master/src/main/scala/com/aliyun/emr/rss/service/deploy/master/Master.scala]: How about current?

File: server-worker/src/main/scala/com/aliyun/emr/rss/service/deploy/worker/Worker.scala [hunk 3]
```diff
@@ -286,7 +286,25 @@ private[deploy] class Worker(
...
+        Thread.sleep(interval)
+        waitTimes += 1
+      }
+      if (partitionLocationInfo.isEmpty) {
+        logInfo(s"Waiting for all PartitionLocation released cost ${waitTime}ms.")
+      } else {
+        logWarning(s"Waiting for all PartitionLocation release cost ${waitTime}ms, " +
+          s"unreleased PartitionLocation: \n$partitionLocationInfo")
+      }
+    }
```
Comment (AngersZhuuuu) [inline server-worker/src/main/scala/com/aliyun/emr/rss/service/deploy/worker/Worker.scala:307]: Worker heartbeat failed can use graceful shutdown too. cc @FMX @waitinfuture

**Review by waitinfuture — APPROVED** [review]
LGTM, thanks!

**Pull request merged**